src/fsort.c

```diff
@@ -43,13 +43,10 @@ static void dradix_r(  // single-threaded recursive worker
     return;
   }
 
-  uint64_t cumSum=0;
-  for (uint64_t i=0; cumSum<n; ++i) { // cumSum<n better than i<width as may return early
-    uint64_t tmp;
-    if ((tmp=counts[i])) {  // don't cumulate through 0s, important below to save a wasteful memset to zero
-      counts[i] = cumSum;
-      cumSum += tmp;
-    }
+  for (uint64_t i = 0, cumSum = 0; cumSum < n; ++i) { // cumSum<n better than i<width as may return early
```

Comment:
It looks like these regions are not covered by existing tests:

https://app.codecov.io/gh/Rdatatable/data.table/pull/7239?dropdown=coverage&src=pr&el=h1

Would you mind devising some cases that would actually use this code? It would be a lot easier to review if we have better assurance there's no behavior change.